src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp

```diff
@@ -1705,18 +1708,26 @@ std::vector<double> EclipseGrid::createDVector(const std::array<int,3>& dims, st
         // create coord vector of floats with input units, converted from SI
         std::vector<float> coord_f;
         coord_f.resize(m_coord.size());
-
-        for (size_t n=0; n< m_coord.size(); n++){
-            coord_f[n] = static_cast<double>(units.from_si(length, m_coord[n]));
-        }
-
+        
+        if (m_input_coord.has_value())
+            for (size_t n=0; n< m_input_coord.value().size(); n++)
+                coord_f[n] = static_cast<double>(units.from_si(length, m_input_coord.value()[n]));
+        else
+            for (size_t n=0; n< m_coord.size(); n++)
+                coord_f[n] = static_cast<double>(units.from_si(length, m_coord[n]));
```

Comment:
Did you mean `static_cast<float>(...)` instead of `static_cast<double>(...)` here?  The values are already `double` but you're assigning to a `float`-valued vector.  For what it's worth, I think I'd probably use `std::transform()` instead of a raw loop
```C++
auto convert_length = [&units](const double x) { return static_cast<float>(units.from_si(length, x); };

if (m_input_coord.has_value()) {
    std::transform(m_input_coord.value().begin(), m_input_coord.value().end(), coord_f.begin(), convert_length);
}
else {
    std::transform(m_coord.begin(), m_icoord.end(), coord_f.begin(), convert_length);
}
```
and similar for `ZCORN`.

Reply:
Yes it should be static_cast<float>. This error is actually from the current version (master). [link](https://github.com/OPM/opm-common/blob/a154a2a49f696d30e617285c8d9897ae8714d8ae/src/opm/input/eclipse/EclipseState/Grid/EclipseGrid.cpp#L1710)

I'm happy to switch to transform instead of the for loop.

Reply:
As we talked about offline yesterday, the underlying issue is most likely related to MINPV/PINCH handling in opm-grid. 
From our discussions we agreed to keep the solution used in this PR for now. Releasing the memory after saving the EGRID file is done.